.github/workflows/helpers.js

```diff
@@ -3,6 +3,62 @@
 const owner = process.env.REPO_OWNER || ''; // example owner: adobecom
 const repo = process.env.REPO_NAME || ''; // example repo name: milo
 const auth = process.env.GH_TOKEN || ''; // https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens
+const CURRENT_YEAR = 2024;
+const RCPDates = [
+  {
+    start: new Date('2024-05-26T00:00:00-07:00'),
+    end: new Date('2024-06-01T00:00:00-07:00'),
+  },
+  {
+    start: new Date('2024-06-13T11:00:00-07:00'),
+    end: new Date('2024-06-13T14:00:00-07:00'),
+  },
+  {
+    start: new Date('2024-06-30T00:00:00-07:00'),
+    end: new Date('2024-07-06T00:00:00-07:00'),
+  },
+  {
+    start: new Date('2024-08-25T00:00:00-07:00'),
+    end: new Date('2024-08-31T00:00:00-07:00'),
+  },
+  {
+    start: new Date('2024-09-12T11:00:00-07:00'),
+    end: new Date('2024-09-12T14:00:00-07:00'),
+  },
+  {
+    start: new Date('2024-10-14T00:00:00-07:00'),
+    end: new Date('2024-11-18T17:00:00-08:00'),
+  },
+  {
+    start: new Date('2024-11-17T00:00:00-08:00'),
+    end: new Date('2024-11-30T00:00:00-08:00'),
+  },
+  {
+    start: new Date('2024-12-12T11:00:00-08:00'),
+    end: new Date('2024-12-12T14:00:00-08:00'),
+  },
+  {
+    start: new Date('2024-12-15T00:00:00-08:00'),
+    end: new Date('2025-01-02T00:00:00-08:00'),
+  },
+];
+
+const isWithinRCP = () => {
+  const now = new Date();
+  if (now.getFullYear() !== CURRENT_YEAR) {
+    console.log(`ADD NEW RCPs for ${CURRENT_YEAR + 1}`);
+    return true;
+  }
+
+  for (const { start, end } of RCPDates) {
+    if (start <= now && now <= end) {
+      console.log('Current date is within a RCP. Stopping execution.');
+      return true;
+    }
+  }
+
+  return false;
```

Comment:
Nit: this can be written shorter
```suggestion
  return RCPDates.some(({ start, end }) => start <= now && now <= end)
    ? (console.log('Current date is within a RCP. Stopping execution.'), true) : false;
```
Feel free to not make this change if you don't want to. It's less code, but might be less readable as it uses the less common [JS comma operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comma_operator)